### PR TITLE
Update wifibroadcast article

### PR DESCRIPTION
**Pull request by svpcom**
Sync article with latest wifibroadcast release

### CODE REVIEW

**Comment by svpcom** [comment]
Any chances to review on this week?

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 1]
```diff
@@ -1,99 +1,118 @@
 # Long-distance video streaming in QGroundControl
 
-This page shows how to set up a a companion computer (Odroid C1 or C0) with a camera (Logitech C920) such that the video stream is transferred via the Odroid C1 to a network computer and displayed in the application QGroundControl that runs on this computer. This setup uses WiFi in unconnected (broadcast) mode.
+This page shows how to set up a a companion computer with a camera (Logitech C920 or RaspberryPI camera) such that the video stream is transferred from UAV to a ground computer and displayed in the application QGroundControl that runs on this computer. This setup uses WiFi in unconnected (broadcast) mode.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: Some typos, replace text with:

This page shows how to set up a companion computer with a camera (Logitech C920 or RaspberryPi camera) such that the video stream is transferred from the UAV to a ground computer and displayed in the QGroundControl application. This setup uses WiFi in unconnected (broadcast) mode.

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 2]
```diff
@@ -1,99 +1,118 @@
...
-* Logitech camera C920
+On TX (UAV) side:
+* NanoPI NEO2 (and/or Raspberry PI if use PI camera)
+* Logitech camera C920 or RaspberryPI camera
 * WiFi module  ALPHA AWUS051NH v2.
 
 On RX (group station side):
 * Any computer with Linux.
-* WiFi module  ALPHA AWUS051NH v2.
+* WiFi module with Ralink RT5572 chipset (CSL 300Mbit Stick or GWF-4M02)
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: Links to good versions people can purchase?

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 3]
```diff
@@ -1,99 +1,118 @@
...
 
 The whole hardware setup consists of the following parts:
 
-On TX (copter) side:
-* Odroid C1
-* Logitech camera C920
+On TX (UAV) side:
+* NanoPI NEO2 (and/or Raspberry PI if use PI camera)
+* Logitech camera C920 or RaspberryPI camera
 * WiFi module  ALPHA AWUS051NH v2.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: Purchase link?

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 4]
```diff
@@ -1,99 +1,118 @@
...
-This page shows how to set up a a companion computer (Odroid C1 or C0) with a camera (Logitech C920) such that the video stream is transferred via the Odroid C1 to a network computer and displayed in the application QGroundControl that runs on this computer. This setup uses WiFi in unconnected (broadcast) mode.
+This page shows how to set up a a companion computer with a camera (Logitech C920 or RaspberryPI camera) such that the video stream is transferred from UAV to a ground computer and displayed in the application QGroundControl that runs on this computer. This setup uses WiFi in unconnected (broadcast) mode.
 
 The whole hardware setup consists of the following parts:
 
-On TX (copter) side:
-* Odroid C1
-* Logitech camera C920
+On TX (UAV) side:
+* NanoPI NEO2 (and/or Raspberry PI if use PI camera)
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: "Raspberry Pi" (Pi is lower case everywhere). Links to manufacturer for components always useful for readers.

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 5]
```diff
@@ -1,99 +1,118 @@
...
 On RX (group station side):
 * Any computer with Linux.
-* WiFi module  ALPHA AWUS051NH v2.
+* WiFi module with Ralink RT5572 chipset (CSL 300Mbit Stick or GWF-4M02)
+
+
+## Wifibroadcast
 
+ - 1:1 map RTP to IEEE80211 packets for minimum latency (doesn't serialize to byte steam)
+ - Smart FEC support (immediately yeild packet to video decoder if FEC pipeline without gaps)
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: spelling: yield

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 6]
```diff
@@ -120,44 +139,60 @@ A: This is can be due to:
...
+
+   Pros:
+     * ARM 64-bit CPU
+     * Very cheap
+     * Supported by mainline kernel
+     * 3 independent USB busses
+     * 1Gbps ethernet port
+     * 3 UARTs
+     * Very small form-factor
+     * Resistent to radio interference
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: spelling: Resistant

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 7]
```diff
@@ -1,99 +1,118 @@
...
+* WiFi module with Ralink RT5572 chipset (CSL 300Mbit Stick or GWF-4M02)
+
+
+## Wifibroadcast
 
+ - 1:1 map RTP to IEEE80211 packets for minimum latency (doesn't serialize to byte steam)
+ - Smart FEC support (immediately yeild packet to video decoder if FEC pipeline without gaps)
+ - Stream encryption and authentication ([libsodium](https://download.libsodium.org/doc/))
+ - Distributed operation. It can gather data from cards on different hosts. So you don't limited to bandwidth of single USB bus.
+ - Aggreagation of mavlink packets. Doesn't send wifi packet for every mavlink packet.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: spelling: Aggregation. 
format: MAVLink

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 8]
```diff
@@ -1,99 +1,118 @@
...
  - Wifibroadcast uses Forward Error Correction to archive a high reliability at low bandwidth requirements. It is able to repair lost or corrupted packets at the receiver.
 
 
+
 ## Hardware modification.
-Alpha WUS051NH is a high power card and eats too much current while TX. If you power it from USB will reset port on Odroid C1/C0.
+Alpha WUS051NH is a high power card and eats too much current while TX. If you power it from USB will reset port on most ARM boards.
 So you need to connect it to 5V BEC directly. You can do this two ways:
 
  1. Make a custom usb cable.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: Any instructions, images, links for making a custom cable.

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 9]
```diff
@@ -1,99 +1,118 @@
...
- 2. Cut a 5V wire on odroid PCB near usb port and wire it to BEC.
+ 2. Cut a 5V wire on PCB near usb port and wire it to BEC.
     Also I suggest to add 470uF low ESR capacitor (like ESC has) between power and ground to filter voltage spikes.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: Circuit diagram/snapshots - something to make it very easy for people to see exactly what you mean and duplicate your work?

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 10]
```diff
@@ -1,99 +1,118 @@
...
-On TX (copter) side:
-* Odroid C1
-* Logitech camera C920
+On TX (UAV) side:
+* NanoPI NEO2 (and/or Raspberry PI if use PI camera)
+* Logitech camera C920 or RaspberryPI camera
 * WiFi module  ALPHA AWUS051NH v2.
 
 On RX (group station side):
 * Any computer with Linux.
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: Any Linux? Is there a recommended/version you tested this on?

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 11]
```diff
@@ -1,99 +1,118 @@
...
+## Wifibroadcast
 
+ - 1:1 map RTP to IEEE80211 packets for minimum latency (doesn't serialize to byte steam)
+ - Smart FEC support (immediately yeild packet to video decoder if FEC pipeline without gaps)
+ - Stream encryption and authentication ([libsodium](https://download.libsodium.org/doc/))
+ - Distributed operation. It can gather data from cards on different hosts. So you don't limited to bandwidth of single USB bus.
+ - Aggreagation of mavlink packets. Doesn't send wifi packet for every mavlink packet.
+ - Enhanced OSD https://github.com/svpcom/wifibroadcast_osd for Raspberry PI (consume 10% CPU on PI Zero)
 
-## Why normal wifi is a bad choice for long-distance video transfer
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: Any particular reason why you chose to remove this information?

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 12]
```diff
@@ -1,99 +1,118 @@
...
-Download wifibroadcast [sources](https://github.com/svpcom/wifibroadcast).
 
-You need to patch kernel to:
- 
- 1. Enable TX rate lock. Use ``mac80211-radiotap-bitrate_mcs_rtscts.linux-4.4.patch``. Instead there are no way to specify data rate for injected radiotap packets.
- 2. Enable TX power lock. Use ``ez-wifibroadcast-1.4-kernel-4.4-patches.diff``. This will lock tx power to maximum supported by card.
- 3. Enable RX of frames with bad FSC (checksum). Use ``ez-wifibroadcast-1.4-kernel-4.4-patches.diff``. This is optional and don't use in current code.
+1. Install libpcap and libsodium development libs.
+2. Download wifibroadcast [sources](https://github.com/svpcom/wifibroadcast).
+3. [Patch](https://github.com/svpcom/wifibroadcast/wiki/Kernel-patches) your kernel. You only need to patch kernel on TX (except if you want to use WiFi channel which is disabled in your region by CRDA).
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: Are there any instructions/links on how to do this patching? The link states what the patches are, but would be great to have starting point for users wanting to try this.

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 13]
```diff
@@ -1,99 +1,118 @@
...
 ```
 gst-launch-1.0 udpsrc port=5600 caps='application/x-rtp, media=(string)video, clock-rate=(int)90000, encoding-name=(string)H264' \
              ! rtph264depay ! avdec_h264 ! clockoverlay valignment=bottom ! autovideosink fps-update-interval=1000 sync=false
 ```
 to decode video.
 
+## Enhanced setup with RX antenna array, FPV goggles and OSD
+See [wiki](https://github.com/svpcom/wifibroadcast/wiki/enhanced-setup) article
+
+## TODO
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md]: Can any of this be done :-)

**Comment by hamishwillee** [comment]
Thanks @svpcom /Vasily !
OK, so there are a bunch of relatively minor English and formatting errors, which I propose to fix AFTER this is merged (ie don't worry about fixing them now).

The corrections are absolutely fine, but this could be improved:
- There is another article on [video streaming](https://dev.px4.io/en/qgc/video_streaming.html) in the guide.  It would be good to add a note to differentiate them so that people can tell which one they should use.
   - Is the main difference using wifibroadcasting? 
   - What is the relative/approximate distance and improvement that can be achieved using this approach?
- When you cut out the topic "## Why normal wifi is a bad choice for long-distance video transfer" it kind of removed the logical flow/introduction to wifibroadcast and theory. What I suggest you do is merge the headings "Wifibroadcast" and "Theory" and precede all those bullet points with something like the first paragraph here: https://befinitiv.wordpress.com/wifibroadcast-analog-like-transmission-of-live-video-data/
- There are some cases with instructions to set up hardware. You explain what needs to happen, but could make the material more accessible/turnkey with:
  - Links to the components you actually tried. I see some of that below but might be worth merging in top section.
- Additional setup info on what you did. E.g. circuit diagrams, images of connections etc.
- Is there any particular setup required for QGC? Can we link to that?

**Comment by svpcom** [comment]
@hamishwillee 
I've fixed all typos and correct article:
 - Added main difference for wifibroadcast vs normal wifi
 - Added measured link distance
 - Added link to USB cable modification.
 - Added link to enhanced setup (distributed RX grid, telemetry, qgroundcontrol, etc)

File: en/qgc/video_streaming_wifi_broadcast.md [hunk 14]
```diff
@@ -1,110 +1,139 @@
...
-On TX (copter) side:
-* Odroid C1
-* Logitech camera C920
-* WiFi module  ALPHA AWUS051NH v2.
+On TX (UAV) side:
+* [NanoPI NEO2](http://www.friendlyarm.com/index.php?route=product/product&product_id=180) (and/or Raspberry Pi if use Pi camera).
+* [Logitech camera C920](https://www.logitech.com/en-us/product/hd-pro-webcam-c920?crid=34) or [Raspberry Pi camera](https://www.raspberrypi.org/products/camera-module-v2/).
+* WiFi module  [ALPHA AWUS051NH v2](https://www.alfa.com.tw/products_show.php?pc=67&ps=241).
 
 On RX (group station side):
```
Comment (hamishwillee) [inline en/qgc/video_streaming_wifi_broadcast.md:12]: Group station? I think you mean ground station.

**Comment by hamishwillee** [comment]
Thanks very much @svpcom ! Merged.

**Comment by hamishwillee** [comment]
I have added my updates in #327. Can you please check those out?

**Pull request merged**